### PR TITLE
br/pkg/version: migrate test-infra to testify

### LINKED ISSUE #28184: migrate test-infra to testify for `br/pkg/version`

### PR DESCRIPTION
**Pull request by dengqee**
### What problem does this PR solve?

Issue Number: close #28184

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/version/version_test.go [hunk 1]
```diff
@@ -157,127 +158,123 @@ func (s *checkSuite) TestCheckClusterVersion(c *C) {
...
 
-func (s *checkSuite) TestCheckVersion(c *C) {
-	err := CheckVersion("TiNB", *semver.New("2.3.5"), *semver.New("2.1.0"), *semver.New("3.0.0"))
-	c.Assert(err, IsNil)
+func TestCheckVersion(t *testing.T) {
+	t.Parallel()
 
-	err = CheckVersion("TiNB", *semver.New("2.1.0"), *semver.New("2.3.5"), *semver.New("3.0.0"))
-	c.Assert(err, ErrorMatches, "TiNB version too old.*")
+	err := CheckVersion("TiDB", *semver.New("2.3.5"), *semver.New("2.1.0"), *semver.New("3.0.0"))
```
Comment (kennytm) [inline br/pkg/version/version_test.go:250]: using "TiNB" here is intentional.

File: br/pkg/version/version_test.go [hunk 2]
```diff
@@ -292,13 +289,15 @@ func (s *checkSuite) TestNormalizeBackupVersion(c *C) {
 	for _, testCase := range cases {
 		target, _ := semver.NewVersion(testCase.target)
 		source := NormalizeBackupVersion(testCase.source)
-		c.Assert(source, versionEquals, target)
+		result := versionEqualCheck(source, target)
+		require.Equal(t, true, result)
```
Comment (kennytm) [inline br/pkg/version/version_test.go:293]:
```suggestion
		require.Truef(t, result, "source=%v, target=%v", source, target)
```

File: br/pkg/version/version_test.go [hunk 3]
```diff
@@ -316,22 +315,23 @@ func (s *checkSuite) TestDetectServerInfo(c *C) {
...
+		require.Equal(t, expectVer == nil, info.ServerVersion == nil, cmt)
 		if info.ServerVersion == nil {
-			c.Assert(expectVer, IsNil, cmt)
+			require.Nil(t, expectVer, cmt)
 		} else {
-			c.Assert(info.ServerVersion.Equal(*expectVer), IsTrue)
+			require.True(t, info.ServerVersion.Equal(*expectVer))
 		}
-		c.Assert(mock.ExpectationsWereMet(), IsNil, cmt)
+		require.Nil(t, mock.ExpectationsWereMet(), cmt)
```
Comment (kennytm) [inline br/pkg/version/version_test.go:334]:
```suggestion
		require.NoError(t, mock.ExpectationsWereMet(), cmt)
```

**Review by kennytm — COMMENTED** [review]
rest LGTM

**Review by kennytm — APPROVED** [review]

**Review by disksing — APPROVED** [review]

**Pull request merged**